Doc/c-api/float.rst

```diff
@@ -76,3 +76,83 @@ Floating Point Objects
 .. c:function:: double PyFloat_GetMin()
 
    Return the minimum normalized positive float *DBL_MIN* as C :c:type:`double`.
+
+
+Pack and Unpack functions
+=========================
+
+The :mod:`struct` and :mod:`pickle` (at least) modules need an efficient
+platform- independent way to store floating-point values as byte strings. The
+Pack routines produce a bytes string from a C :c:type:`double`, and the Unpack
+routines produce a C :c:type:`double` from such a bytes string. The suffix
+(2, 4 or 8) specifies the number of bytes in the bytes string.
+
+On platforms that appear to use IEEE-754 formats these functions work by
+copying bits. On other platforms, the formats the 4- byte format (32-bit) is
+identical to the IEEE-754 single precision format, and the 8-byte format
+(64-bit) to the IEEE-754 double precision format, although the packing of INFs
+and NaNs (if such things exist on the platform) isn't handled correctly, and
+attempting to unpack a bytes string containing an IEEE INF or NaN will raise an
+exception.
+
+On non-IEEE platforms with more precision, or larger dynamic range, than IEEE
```

Comment:
Presumably once we're assuming IEEE 754 for CPython, all this text can be dropped?

Reply:
My plan is to first fix the Python 3.11 "regression" by exposing these functions as public C API functions. And then deprecate or remove support for non-IEEE 754 platforms.

Reply:
Require IEEE 754 support is tracked by https://bugs.python.org/issue46917